### ISSUE TITLE
The --js_out flag support

### ISSUE THREAD

**Issue by ilya-medvedev** [os72/protoc-jar-maven-plugin#10]
Hello! Please add --js_out flag support. This enable use binary encoder and decoder in js.
Now I can not write so:
```xml
<outputTarget>
    <type>js</type>
    <outputDirectory>binary:${project.build.outputDirectory}/static/js</outputDirectory>
</outputTarget>
```
Details: https://github.com/google/protobuf/blob/v3.0.0/js/README.md

**Comment by os72** [comment]
Type "js" should work, doesn't it?
You also filed this for protobuf-shaded, I'm confused, you want the shading feature for Javascript?

**Comment by ilya-medvedev** [comment]
I mistakenly created the ticket for protobuf-shaded, sorry.
 
Binary encoder and decoder in js working only with special --js_out flag, but if i writing this flag to outputDirectory then protoc-jar-maven-plugin not worked.

**Comment by ilya-medvedev** [comment]
protoc-jar-maven-plugin believes this path is relative and appends project.basedir path to top

**Comment by os72** [comment]
I'm not sure I understand. Following example works fine, produces Java in src/test/java, Javascript in src/test/js:

						<outputTargets>
							<outputTarget>
								<type>java</type>
								<addSources>none</addSources>
								<outputDirectory>src/test/java</outputDirectory>
							</outputTarget>
							<outputTarget>
								<type>js</type>
								<addSources>none</addSources>
								<outputDirectory>src/test/js</outputDirectory>
							</outputTarget>
						</outputTargets>

**Comment by ilya-medvedev** [comment]
But i need to use outputDirectory with "binary" flag for js:
```xml
<outputTargets>
    <outputTarget>
        <type>java</type>
       <addSources>none</addSources>
        <outputDirectory>src/test/java</outputDirectory>
    </outputTarget>
    <outputTarget>
        <type>js</type>
        <addSources>none</addSources>
        <outputDirectory>binary:src/test/js</outputDirectory>
    </outputTarget>
</outputTargets>
```

**Comment by os72** [comment]
Sorry, I'm not familiar with that notation. What is it supposed to do, and why is it needed?

**Comment by ilya-medvedev** [comment]
This annotation enables binary encoder and decoder in js

Details: https://github.com/google/protobuf/blob/v3.0.0/js/README.md
And https://github.com/google/protobuf/blob/master/src/google/protobuf/compiler/js/js_generator.h#L53

**Comment by os72** [comment]
Got it, it's not just a path anymore, it supports: `--js_out=[OPTIONS:]output_dir`
I think it may be better to add `<outputOptions>` parameter for this, what do you think?

**Comment by ilya-medvedev** [comment]
Yes all right: https://github.com/os72/protoc-jar-maven-plugin/pull/11

**Comment by os72** [comment]
Okay, will include in 3.0.0 release

**Comment by ilya-medvedev** [comment]
Thank you

**Comment by os72** [comment]
Could you please pull my changes and give it a try? (commit 72812d0)

Usage example:

							<outputTarget>
								<type>js</type>
								<addSources>none</addSources>
								<outputOptions>import_style=commonjs,binary</outputOptions>
								<outputDirectory>src/test/js</outputDirectory>
							</outputTarget>

**Comment by os72** [comment]
Released 3.0.0

The parameter is `<outputOptions>`
For example: `<outputOptions>import_style=commonjs,binary</outputOptions>`
